components/motor/i2cmotors/ezopmp.go

```diff
@@ -59,7 +59,7 @@ const modelName = "ezopmp"
 func init() {
 	_motor := registry.Component{
 		Constructor: func(ctx context.Context, deps registry.Dependencies, config config.Component, logger golog.Logger) (interface{}, error) {
-			return NewMotor(ctx, deps, config.ConvertedAttributes.(*AttrConfig), logger)
+			return NewMotor(ctx, deps, config.ConvertedAttributes.(*AttrConfig), config, logger)
```

Comment:
Yeah, if the two configuration objects passed to `NewMotor` originate from a single source config, I suspect we can modify `config.ConvertedAttributes` to store the name and then we don't need the extra argument to `NewMotor`.

Reply:
I think that's a much deeper change that would alter all of our other components as well, instead I think name should be passed as a string argument to NewMotor

Reply:
Gautham has a more informed opinion than I do. Let's leave this as-is!